packit/base_git.py

```diff
@@ -414,3 +421,51 @@ def push(self, refspec: str, remote_name: str = "origin", force: bool = False):
                 raise PackitException(
                     f"We were unable to push to dist-git: {pi.summary}."
                 )
+
+    def download_remote_sources(self):
+        """
+        Download the sources from the URL in the configuration (if the path in
+        the configuration match to the URL basename from SourceX) or from the one
+        from SourceX in specfile.
+        """
+        # filter out only sources with URL
+        remote_files = [
+            source
+            for source in self.specfile.sources
+            if bool(urllib.parse.urlparse(source).scheme)
+        ]
+
+        # download any sources that are not yet downloaded
+        for remote_file in remote_files:
+            basename = os.path.basename(remote_file)
+            local_file = os.path.join(self.specfile.sources_location, basename)
+            if not os.path.isfile(local_file):
+                logger.info(f"File {local_file} doesn't exist locally, downloading it.")
+                url = self.get_matching_source_url_from_config(basename) or remote_file
+                try:
+                    DownloadHelper.download_file(url, local_file)
+                except DownloadError as e:
+                    raise RebaseHelperError(
+                        f"Failed to download file from URL {remote_file}. "
+                        f"Reason: {e}. "
+                    ) from e
+
+    def get_matching_source_url_from_config(self, source_file) -> Optional[str]:
+        """
+        Try to find a source item in the config with the same name as source_file
+        and return its URL.
+        :param source_file: source file to find in the config
+        :return: URL
+        """
+        matching = [
+            source.url
+            for source in self.package_config.sources
+            if source.path == source_file
+        ]
+        if matching:
+            logger.debug(
+                f"Found matching URL for {source_file} in the configuration: {matching}."
+                "Downloading from the first one."
```

Comment:
This is somewhat missleading, as downloading doesn't happen in this method. How about removing this seconds sentence and log `url` and `local_file` before the `DownloadHelper` call?

Reply:
changed